rxgroups/src/main/java/com/airbnb/rxgroups/ResubscribeHelper.java

```diff
@@ -9,54 +9,66 @@
 
 class ResubscribeHelper {
 
-    private static final Map<Class<?>, Constructor<?>> BINDINGS = new LinkedHashMap<>();
-
-    static void initializeResubscription(Object target, ObservableGroup group) {
-        Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
-        if (constructor == null) {
-            throw new IllegalArgumentException("Invalid target: " + target);
-        }
-
-        try {
-            constructor.newInstance(target, group);
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException("Unable to invoke " + constructor, e);
-        } catch (InstantiationException e) {
-            throw new RuntimeException("Unable to invoke " + constructor, e);
-        } catch (InvocationTargetException e) {
-            Throwable cause = e.getCause();
-            if (cause instanceof RuntimeException) {
-                throw (RuntimeException) cause;
-            }
-            if (cause instanceof Error) {
-                throw (Error) cause;
-            }
-            throw new RuntimeException("Unable to create resubscribe instance.", cause);
-        }
+  private static final Map<Class<?>, Constructor<?>> BINDINGS = new LinkedHashMap<>();
+
+  static void initializeResubscription(Object target, ObservableGroup group) {
+    Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
+    if (constructor == null) {
+      throw new IllegalArgumentException("Invalid target: " + target);
+    }
+    invokeConstructor(constructor, target, group);
+  }
+
+  static void safeInitializeResubscription(Object target, ObservableGroup group) {
+    Constructor<?> constructor = findConstructorForClass(target.getClass(), group);
+    if (constructor == null) {
+      return;
+    }
+    invokeConstructor(constructor, target, group);
+  }
+
+  private static void invokeConstructor(Constructor<?> constructor, Object target,
+                                        ObservableGroup group) {
+    try {
+      constructor.newInstance(target, group);
+    } catch (IllegalAccessException e) {
+      throw new RuntimeException("Unable to invoke " + constructor, e);
+    } catch (InstantiationException e) {
+      throw new RuntimeException("Unable to invoke " + constructor, e);
+    } catch (InvocationTargetException e) {
+      Throwable cause = e.getCause();
+      if (cause instanceof RuntimeException) {
+        throw (RuntimeException) cause;
+      }
+      if (cause instanceof Error) {
+        throw (Error) cause;
+      }
+      throw new RuntimeException("Unable to create resubscribeAll instance.", cause);
+    }
+  }
+
+  private static Constructor<?> findConstructorForClass(Class<?> cls, ObservableGroup group) {
+    Constructor<?> bindingCtor = BINDINGS.get(cls);
+    if (bindingCtor != null) {
```

Comment:
with this similar pattern in other annotation processor uses I've found that many classes legitimately don't have a constructor, but since our base AirFragment always tries to initialize it the `null` value is never stored. we can optimize this by storing null in the map, and then checking `if (bindingCtor != null || BINDINGS.get(cls))`. This helps in the case that a fragment is loaded many times but doesn't have a constructor (in our case, a fragment with no network requests), it doesn't have to do this reflection every time